server/src/main/java/org/elasticsearch/index/query/InnerHitContextBuilder.java

```diff
@@ -77,43 +79,53 @@ public static void extractInnerHits(QueryBuilder query, Map<String, InnerHitCont
         }
     }
 
-    protected void setupInnerHitsContext(QueryShardContext queryShardContext,
-                                         InnerHitsContext.InnerHitSubContext innerHitsContext) throws IOException {
-        innerHitsContext.from(innerHitBuilder.getFrom());
-        innerHitsContext.size(innerHitBuilder.getSize());
-        innerHitsContext.explain(innerHitBuilder.isExplain());
-        innerHitsContext.version(innerHitBuilder.isVersion());
-        innerHitsContext.seqNoAndPrimaryTerm(innerHitBuilder.isSeqNoAndPrimaryTerm());
-        innerHitsContext.trackScores(innerHitBuilder.isTrackScores());
+    protected SearchContext createSubSearchContext(QueryShardContext cloneShardContext, SearchContext parentContext) throws IOException {
```

Comment:
I tried a different approach in https://github.com/elastic/elasticsearch/pull/47733. We don't need a full search context to run the fetch phase or the highlight phase. I started with the highlight phase but it could be applicable to the fetch phase too and in this case we wouldn't need a sub search context.